### ISSUE TITLE
(javascript) certain expression breaks future highlighting

### ISSUE THREAD

**Issue by paladox** [highlightjs/highlight.js#2258]
Hi, it appears that doing:

```js
    _awaitOpen(fn) {
      let iters = 0;
      const step = () => {
        this.async(() => {
          if (this.$.dropdown.style.display !== 'none') {
            fn.call(this);
          } else if (iters++ < AWAIT_MAX_ITERS) {
            step.call(this);
          }
        }, AWAIT_STEP);
      };
      step.call(this);
    },
```

Causes the syntax after  `< AWAIT_MAX_ITERS` to not be highlighted. (so everything and below).

here's a jsfiddle https://jsfiddle.net/ajoshguy/h4v57csf/ to show the issue.  `_id` function return should be highlighted.

**Comment by paladox** [comment]
Oh, yes, sorry, thanks. Updated description.

**Comment by yyyc514** [comment]
Closing as this seems fixed in 9.18.0 (at least).
